### PR TITLE
Fix fuzzer when only explicit faults are used

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fix fuzzer when only explicit faults are used

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

<details>

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 33885b27bfba856410ee18332f37489db753f228 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63775/33885b27bfba856410ee18332f37489db753f228/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Common/ThreadFuzzer.cpp [hunk 1]
```diff
@@ -146,7 +153,7 @@ bool ThreadFuzzer::isEffective() const
             return true; \
         if (NAME##_before_sleep_probability.load(std::memory_order_relaxed) > 0.0) \
             return true; \
-        if (NAME##_before_sleep_time_us_max.load(std::memory_order_relaxed) > 0.0) \
+        if (NAME##_before_sleep_time_us_max.load(std::memory_order_relaxed) > 0.001) \
```
Comment (alexey-milovidov) [inline src/Common/ThreadFuzzer.cpp:156]: This looks strange to me.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by alexey-milovidov** [comment]
Only a user-visible misbehavior in an official stable release for a production feature qualifies as a bug fix.

File: src/Common/ThreadFuzzer.cpp (hunk 1 above)
Comment (Algunenano) [reply src/Common/ThreadFuzzer.cpp:156]: I'll change it to check it in the `sleepForNanoseconds` call instead. The problem comes from trying to cast to an integer and calculate a mod % value

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**